docs/Music Macro Language.md

```diff
@@ -0,0 +1,989 @@
+| 08 | testchan |    |    |    | stopchan | subio | ldres | stio | ldio | ldchan | rldchan | ldseq | sub       |             |    | *loop         |
+| 09 | testchan |    |    |    | stopchan | subio | ldres | stio | ldio | ldchan | rldchan | ldseq | and       | fadescale   |    | *bltz         |
+| 0A | testchan |    |    |    | stopchan | subio | ldres | stio | ldio | ldchan | rldchan | ldseq |           | fade        |    | *beqz         |
+| 0B | testchan |    |    |    | stopchan | subio | ldres | stio | ldio | ldchan | rldchan | ldseq |           | fadetovol   |    | *jump         |
+| 0C | testchan |    |    |    | stopchan | subio | ldres | stio | ldio | ldchan | rldchan | ldseq | ldi       | tempochg    |    | *call         |
+| 0D | testchan |    |    |    | stopchan | subio | ldres | stio | ldio | ldchan | rldchan | ldseq | dyncall   | tempo       |    | *delay        |
+| 0E | testchan |    |    |    | stopchan | subio | ldres | stio | ldio | ldchan | rldchan | ldseq | rand      | rtranspose  |    | *delay1       |
+| 0F | testchan |    |    |    | stopchan | subio | ldres | stio | ldio | ldchan | rldchan | ldseq |           | transpose   |    | *end          |
+
+Instructions marked with an asterisk (\*) are documented in the section *Branching Instructions*.
+
+**testchan** *N*
+
+**Description:** Stores whether or not channel *N* is currently active in `$V`.  If the channel is active, `$V` will equal `0`, otherwise it will equal `1`.
+
+**Parameters:**
+* `N` [integer: 0-15] - The channel number to test.
+
+**Example:**
+```
+.sequence wait_until_done
+delay1
+testchan 1
+beqz wait_until_done
+# Instructions here only execute once channel 1 reaches an end command.
+```
+
+**stopchan** *N*
+
+**Description:** Stops channel *N* from executing any further instructions.  Any notes currently playing on the channel will cease.
+
+**Parameters:**
+* `N` [integer: 0-15] - The channel number to stop.
+
+**Example:**
+```
+ldchan 0, chan0
+ldchan 1, chan1
+delay 50
+stopchan 1      # Stops channel 1 after 50 ticks, even if there's additional instructions.
+```
+
+**subio** *N*
+
+**Description:** Subtracts from `$V` whatever value is in IO port *N*.
+
+**Parameters:**
+* `N` [integer: 0-7] - The IO port to read.
+
+**Example:**
+```
+# Assume that the current value in $V is 5 and the current value in IO port 1 is -5.
+subio 1
+stio 0
+# IO port 0 now contains the current value of $V, which is 0.
+```
+
+**ldres** *IO*, *type*, *index*
+
+**Description:** Loads an audio resource into memory asynchronously.  The type of resource is specified via *type*, where 0 is a sequence, 1 is a soundfont, and 2 is a sample bank.  The IO parameter is used to track the status of the loading process.  The audio subsystem will update this value to 0 once the resource has been loaded.  Note that the IO port doesn't change this value to anything but 0 first, so it's recommended to set the IO port to a non-zero value prior to using `ldres`.
```

Comment:
"audio resource" is the first time I've heard that term used to describe the collection of data that is `sequence`/`soundfont`/`sample bank`. It does make sense to me, just highlighting that nothing in decomp uses that term (which could change)

Reply:
TBD CC @MNGoldenEagle @Dragorn421